pkg/manifests/manifests_test.go

```diff
@@ -1176,6 +1177,63 @@ func TestOpenShiftStateMetrics(t *testing.T) {
 	}
 }
 
+func TestPrometheusK8sControlPlaneRulesFiltered(t *testing.T) {
+	tests := []struct {
+		name   string
+		config *Config
+		verify func(bool, bool, bool)
+	}{
+		{
+			name: "default config",
+			config: func() *Config {
+				c := NewDefaultConfig()
+				c.Platform = configv1.AWSPlatformType
+				return c
+			}(),
+			verify: func(api, cm, sched bool) {
+				if !api || !cm || !sched {
+					t.Fatal("did not get all expected kubernetes control plane rules")
+				}
+			},
+		},
+		{
+			name: "hosted control plane",
+			config: func() *Config {
+				c := NewDefaultConfig()
+				c.Platform = IBMCloudPlatformType
+				return c
+			}(),
+			verify: func(api, cm, sched bool) {
+				if api || cm || sched {
+					t.Fatalf("kubernetes control plane rules found, none expected")
+				}
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		f := NewFactory("openshift-monitoring", "openshift-user-workload-monitoring", tc.config)
+		r, err := f.PrometheusK8sRules()
+		if err != nil {
+			t.Fatal(err)
+		}
+		apiServerRulesFound := false
+		controllerManagerRulesFound := false
+		schedulerRulesFound := false
+		for _, g := range r.Spec.Groups {
+			switch g.Name {
+			case "kubernetes-system-apiserver":
```

Comment:
what about etcd? and what about the servicemonitors that are provisioned to monitor these in the first place?

Reply:
Service monitors that are part of control plane operators are skipped via annotation: https://github.com/openshift/cluster-kube-apiserver-operator/blob/4de583dffa47c25adde871e7361386b0d55c674a/manifests/0000_90_kube-apiserver-operator_03_servicemonitor.yaml#L7
Etcd rules are disabled because the `etcd-metric-client` secret is not placed in the `openshift-config` namespace: https://github.com/openshift/cluster-monitoring-operator/blob/207ec4ea710aa7397dd9cd578d09b4d4a8606074/pkg/operator/operator.go#L418-L421

Reply:
I see ok.